### DISCUSSION TITLE
Passing props using router-view

### DISCUSSION THREAD

**Discussion by bxia-paul** [quasarframework/quasar#12879] [General - Components / Directives / etc] [answered]
I would like to use router-view to dynamically switch components however, I would also like to be able to pass a prop that updates when the user clicks on a tab.
If I do it below, it passes the prop but I'm not able to view other components on the same layout (I can't switch components).
```
<q-page-container  class="bg-grey-2"> 
       <post-board :tabText="tagText"/>
   </q-page-container>
```

However, when I do it this way, I can view the different components but I just can't pass a prop to the component.
```
<q-page-container  class="bg-grey-2"> 
       <router-view />
   </q-page-container>
```
Any help will be appreciated

**Comment by yusufkandemir** [comment, marked as answer]
You can just pass the props you want to `router-view`, and it would pass them to the actual route component(e.g. `post-board`).
```html
<router-view :tabText="tagText" />
```

Also, there **might** be better alternatives depending on the actual use case. [Provide / Inject](https://vuejs.org/guide/components/provide-inject.html) and [State Management](https://vuejs.org/guide/scaling-up/state-management.html) are alternatives to passing stuff around with props. You can also read about them to see if they fit your case better.

**Reply by bxia-paul** [reply]
Thank you very much for your help, I was able to get it to work using Provide/Inject. They do fit my case better. 
However, just for knowledge sake, I saw that the prop was changing when using it in router-view, however, I was not able to access it.
Do you know how to do this?

```
console.log(props.tabText)
console.log(route.params.tabText)
```

I tried the following code, but it said undefined each time.

**Reply by metalsadman** [reply]
https://router.vuejs.org/guide/essentials/passing-props.html
